weave/integrations/bedrock/bedrock_sdk.py

```diff
@@ -257,7 +409,22 @@ def _patch_converse_stream(bedrock_client: "BaseClient") -> None:
 
 
 def patch_client(bedrock_client: "BaseClient") -> None:
-    _patch_converse(bedrock_client)
-    _patch_invoke(bedrock_client)
-    _patch_apply_guardrail(bedrock_client)
-    _patch_converse_stream(bedrock_client)
+    # Check if this is a bedrock-agent-runtime client
+    if hasattr(bedrock_client, "invoke_agent"):
+        _patch_invoke_agent(bedrock_client)
+    elif hasattr(bedrock_client, "converse") or hasattr(bedrock_client, "invoke_model"):
+        # This is a standard bedrock-runtime client
+        _patch_converse(bedrock_client)
+        _patch_converse_stream(bedrock_client)
+        _patch_invoke(bedrock_client)
+    elif hasattr(bedrock_client, "apply_guardrail"):
+        _patch_apply_guardrail(bedrock_client)
```

Comment:
What is this path?

Reply:
They have 2 different clients! Why make things simple.

Reply:
actually 3!